FileDelete/FileDelete/handle.cpp

```diff
@@ -5,21 +5,23 @@ extern "C" NTSTATUS NTAPI ZwQuerySystemInformation(IN size_t SystemInformationCl
     IN ULONG SystemInformationLength,
     OUT PULONG ReturnLength OPTIONAL);
 
-SYSTEM_HANDLE_INFORMATION* get_all_handles() {
+SYSTEM_HANDLE_INFORMATION* get_all_handles()
+{
     size_t handles_allocation_size = 0;
     PVOID handles_pool = nullptr;
 
-    while (true) {
+    for (;;)
+    {
         handles_allocation_size += 0x10000;
-        handles_pool = ExAllocatePool(PagedPool, handles_allocation_size);
+        handles_pool = ExAllocatePool2(PagedPool, handles_allocation_size, '1cba');
```

Comment:
Can you explain please why this is necessary?

Reply:
This is a good question. Generally speaking, Microsoft Windows is updating its document all the time. Judging whether it succeeds in allocating pools is always required. Other changes could be ignored if using an old version wdk. 

According to the Microsoft [document](https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-exallocatepool), ExAllocatePool is obsolete and has been deprecated in Windows 10, version 2004. It has been replaced by [ExAllocatePool2](https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-exallocatepool2). From then on, Microsoft always encourages developers to have a tag while allocating a pool. The following image prevents me to build it when regarding warnings as errors in kernel mode. 

![image](https://user-images.githubusercontent.com/57588843/181788907-2803d0b6-629e-42e6-967f-c13d9302faf6.png)

After my testing, ExAllocatePool2 is not always successful, but ExAllocatePoolZero on my system could succeed. I have no good ideas on this point. Maybe it is better to judge which one to use first according to the system. 
![image](https://user-images.githubusercontent.com/57588843/181789674-e42b3598-f9b0-4026-9142-005e871ec3dc.png)

The same as the ProcessReference.cpp. 
Many Thanks.

Reply:
Cool :)
Why you didn't choose ExAllocatePoolWithTag?
Because [ExAllocatePool2](https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-exallocatepool2) is supported only from new windows version.

Reply:
ExAllocatePoolWithTag is also replaced by ExAllocatePool2. 
I don't know why Microsoft would like to do this. Maybe it is safer to use the new API. 
Click [here](https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/wdm/nf-wdm-exallocatepoolwithtag) to see related documents.

Reply:
ExAllocatePool2 and ExAllocatePoolZero are both functions that are supported from new versions of windows.
ExAllocatePoolWithTag is maybe deprecated in the new WDK toolset but there is background compatibility in the windows kernel, you should compile with WDK from from lower version.
Please use ExAllocatePoolWithTag